standards/git-commands.md

```diff
@@ -1,22 +1,91 @@
 # Git Commands
+
 Git is a distributed version-control system for tracking changes in source code during software development. It is designed for coordinating work among programmers, but it can be used to track changes in any set of files. Its goals include speed, data integrity, and support for distributed, non-linear workflows.
 
 At Bitwise, we use git for a majority of our software projects.
 
 ## Bitwise Git specific resources
+
 1. [Commits](commits.md)
 2. [Branching](branching.md)
 3. [Code Versioning](code-versioning.md)
 
 ### Lesser known Git Commands
 
+#### Git Switch
+
+Git Official Documentation: https://git-scm.com/docs/git-switch
+
+Git Switch (and Restore) were added to address overloading of the `checkout` command, and to provide alternative (and clearer) ways to do some basic work.
+These commands do not provide new functionality, but may be easier to remember how to use them and what they do.
+
+Switch is useful to switch to another branch, creating a new branch in the process if desired.
+
+```shell
+$ git switch <branch-name>
+```
+
+can be used to switch to an existing branch instead of the checkout analog:
+
+```shell
+$ git checkout <branch-name>
+```
+
+For creating new branches, then switching to them, `switch -c` may seem more intuitive than `checkout -b`.
+
+```shell
+$ git switch -c <branch-name>
+```
+
+can be used to create a new branch and switch to it. Alternatively,
+
+```shell
+$ git checkout -b <branch-name>
+```
+
+provides identical functionality.
+
+#### Git Restore
+
+Git Official Documentation: https://git-scm.com/docs/git-restore
+
+Git Switch (and Restore) were added to address overloading of the `checkout` command, and to provide alternative (and clearer) ways to do some basic work.
+These commands do not provide new functionality, but may be easier to remember how to use them and what they do.
+
+Git Restore is useful to undo either changes made or changes staged.
+
+To undo changes made since the last commit (back to HEAD):
+
+```shell
+$ git restore <file-name>
+```
+
+If we wanted to restore the file to what it looked like say, two commits ago, we can use `--source` option and `HEAD~1` pointer:
+
+```text
+$ git restore --source HEAD~1 <file-name>
```

Comment:
Would this be equivalent to something like `git rebase HEAD~2 ` ?

Reply:
These commands get pretty complicated so I hope I can do this justice.

Rebase is a history revision tool, while restore is more like a backup. 
Rebase changes your main tree. Restore changes current branch.

For instance, you can restore just one file from several commits back. ie "Time Travel". It can also be used for multiple files. Restore works from the branch we're at and restores a file or files to a certain point of a respective file's history. 

Rebase is a great way to clean up history, but Its really a totally different beast.

Reply:
Nice thanks for the explanation! Ill be trying it out the next couple of weeks.